tests/test_config_flow.py

```
@@ -0,0 +1,1334 @@
+async def test_get_device_success(hass):
+    mock_client = MagicMock()
+    mock_client.connect = AsyncMock(return_value=AUTHN_SUCCESS)
+    mock_client.get_user_device_data = AsyncMock(
+        return_value={
+            "serialNumber": "123",
+            "fgDeviceManufacturerName": "ResMed",
+            "localizedName": "MyDevice",
+        }
+    )
+    with patch("custom_components.resmed_myair.config_flow.RESTClient", return_value=mock_client):
+        with patch("custom_components.resmed_myair.config_flow.async_create_clientsession"):
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)